lib/ui_components/docu_cop.rb

```diff
@@ -0,0 +1,77 @@
+module UiComponents
+  module DocuCop
+    extend ActiveSupport::Concern
+
+    def initialize(*args)
+      super
+      options.except(:controller).each do |k, v|
+        public_send(:"#{k}=", v)
+      end
+      validate_mandatory_attributes
+    end
+
+    def attributes
+      self.class.attributes.keys.map do |k|
+        [k, public_send(k)]
+      end.to_h
+    end
+
+    private
+
+    def validate_mandatory_attributes
+      missing_arguments = self.class.attributes
+        .select { |a| a[:mandatory] && public_send(a[:name]).nil? }
+      return unless missing_arguments.present?
+      fail MandatoryPropertyNotSet,
```

Comment:
How exactly does this naming scheme work? We have `attributes`, `arguments` and `properties`?